### PR TITLE
Release of GeoNetwork 4.0.4.

### PR DESCRIPTION
**Pull request by fxprunayre**

### CODE REVIEW

File: 4.0.4/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    volumes:
+      - esdata:/usr/share/elasticsearch/data
+    networks:
+      - gn-network
+
+  kibana:
+    image: kibana:7.9.2
+    environment:
+      SERVER_NAME: 'kibana'
+      ELASTICSEARCH_URL: http://elasticsearch:9200/
```
Comment (juanluisrp) [inline 4.0.4/docker-compose.yml:130]: Shouldn't this be `ELASTICSEARCH_HOSTS: http://elasticsearch:9200`? I think from Kibana 7 this is the variable to set.

**Comment by juanluisrp** [comment]
Running the image I see this warning:
```
Password database encryptor initialized - Keep the file /var/lib/jetty/webapps/geonetwork/WEB-INF/data/config/encryptor.properties safe and make a backup. When upgrading to a newer version of GeoNetwork the file must be restored, otherwise GeoNetwork will not be able to decrypt passwords already stored in the database.
```

However the data directory is set to `-Dgeonetwork.data.dir=/catalogue-data`. Is there any way to specify the path of this file so we can put it in the data directory and keep it when the container is destroyed/re-created?
@fxprunayre @josegar74

File: 4.0.4/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    environment:
+      ES_JAVA_OPTS: "-Xms1G -Xmx1G"
+      discovery.type: single-node
+    volumes:
+      - esdata:/usr/share/elasticsearch/data
+    networks:
+      - gn-network
+
+  kibana:
+    image: kibana:7.9.2
```
[the remark this answers is not in the document]
Comment (fxprunayre) [reply 4.0.4/docker-compose.yml:127]: Not really we are not doing advanced stuff with Elastic (I've 4.0.4 deployed on version 7.5.2 to 7.12.0 in various projects). The main reason to update is generally new features in Kibana - so I think we can keep the update for the next one with https://github.com/geonetwork/core-geonetwork/pull/5623

**Comment by juanluisrp** [comment]
In the meantime I can mount the file directly into the container:

```shell
touch encryptor.properties
```

https://github.com/geonetwork/docker-geonetwork/pull/68/files#diff-b696ff9eb7241c62028b666784478b8503796af86ee66c7e3c0d2b9b7c31b79eR99-R101

**Comment by juanluisrp** [comment]
Created PR to official images repo https://github.com/docker-library/official-images/pull/10143.

**Pull request merged**